src/ape/managers/chain.py

```diff
@@ -389,49 +389,77 @@ class AccountHistory(BaseInterfaceModel):
     """
 
     address: AddressType
-    _transaction_list: List[ReceiptAPI] = []
+    _session_receipts: List[ReceiptAPI] = []
 
     @property
     def outgoing(self) -> Iterator[ReceiptAPI]:
         """
-        All outgoing transactions.
+        All outgoing transactions, from earliest to latest.
         """
         explorer = self.provider.network.explorer
         explorer_receipts = (
             [r for r in explorer.get_account_transactions(self.address)] if explorer else []
         )
         for receipt in explorer_receipts:
-            # TODO: Use cache
-            if receipt.txn_hash not in [t.txn_hash for t in self._transaction_list]:
+            if receipt.sender != self.address:
+                # Likely ``self.address`` is a contract.
+                # Cache the receipts by their sender instead and skip them here.
+                self.chain_manager.history.append(receipt)
+                continue
+
+            elif receipt.txn_hash not in [t.txn_hash for t in self._session_receipts]:
                 yield receipt
 
     @property
     def all(self) -> Iterator[ReceiptAPI]:
```

Comment:
One thought I was having is that we probably shouldn't have this property until we legitimately have *all* of the transactions indexed, otherwise it's semantics are gonna change when adding `.incoming`, `.internal`, etc.